app/services/push_notification/account_delete.rb

```diff
@@ -0,0 +1,84 @@
+module PushNotification
+  class AccountDelete
+    def call(user_id)
+      send_updates_to_subscribers(user_id)
+    end
+
+    private
+
+    def push_notify(push_to_url, uuid, agency_id)
+      payload = { uuid: uuid }
+      post_to_push_notification_url(push_to_url, payload)
+    rescue Faraday::TimeoutError, Faraday::ConnectionFailed => exception
+      Rails.logger.error "Push message failed #{exception.message}"
+      PushAccountDelete.create(created_at: Time.zone.now, agency_id: agency_id, uuid: uuid)
+    end
+
+    # :reek:FeatureEnvy
+    def post_to_push_notification_url(push_to_url, payload)
+      adapter = faraday_adapter(push_to_url)
+      adapter.post do |request|
+        headers_hash = request.headers
+        headers_hash['topic'] = 'account_delete'
+        headers_hash['content-type'] = 'application/json'
+        headers_hash['authorization'] = web_push(push_to_url, payload)
+      end
+    end
+
+    def signature(unsigned_token)
+      JWT.encode(unsigned_token, RequestKeyManager.private_key, 'RS256')
+    end
+
+    def web_push(push_to_url, payload)
+      unsigned_token = "#{jwt_info}.#{jwt_data(push_to_url, payload)}"
```

Comment:
For the toke here, you are probable looking for something like this:

```ruby
jwt_data = {
  aud: push_to_url,
  exp: (Time.zone.now + 12.hours).to_i,
  uuid: uuid,
  sub: 'https://secure.login.gov',
}
JWT.encode(jwt_data, RequestKeyManager.private_key, 'RS256')
```

Reply:
The payload namespaces it otherwise you would have to merge the hashes and potentially have a conflict (ie now we are sending uuid but other events might be more attributes)